[MTM-64133] Single sign on, document new idp_hint mechanism (#3774)
SoftwareAG/c8y-docs: content/authentication/sso-bundle/custom-template.md

### Placeholders {#placeholders}
Inside some fields you can use placeholders that are resolved by {{< product-c8y-iot >}} at runtime. Available placeholders are:

| Placeholder  | Description                                                                    |
|:-------------|:-------------------------------------------------------------------------------|
| clientId     | Value of the **Client ID** field                                               
| redirectUri  | Value of the **Redirect URI** field                                            
| code         | Code returned by the authorization server in response to authorization request 
| refreshToken | Refresh token returned by the authorization server after token request         

> | id_token     | A jwt token issued by the Auth Server, that contains claims about the authenticated user, can be used for logout in Auth Server

Reviewer: ```suggestion
| id_token     | A JWT token issued by the authorization server that contains claims about the authenticated user; can be used for logout from the authorization server
```

> | idp_hint     | Value of the **idp_hint** parameter passed by UI aplication, to choose which IdP should be preselected during the authentication flow.                    

Reviewer: ```suggestion
| idp_hint     | Value of the `idp_hint` parameter passed by the UI application, to select which IdP should be preselected during the authentication flow                   
```

{{< c8y-admon-info >}}
Placeholders are not validated for correctness. Any not recognized or misspelled placeholder will be left in text unprocessed.
{{< /c8y-admon-info >}}

> ### Client Suggested Identity Provider

Reviewer: ```suggestion
### Client suggested identity provider {#client-suggested-identity-provider}
```

> When the default login mode is **Single sign-on redirect**, and the Auth Server supports multiple identity providers, instead of displaying an intermediate "select identity provider" screen to the user, the `idp_hint` parameter can be collected by {{< product-c8y-iot >}} and forwarded to the Auth Server as part of the initial authorization request. This ensures that the user is redirected directly to the specified IdP.

Reviewer: ```suggestion
When the default login mode is **Single sign-on redirect**, and the authentication server supports multiple identity providers, instead of requesting the user to select an identity provider, the `idp_hint` parameter can be collected by {{< product-c8y-iot >}} and forwarded to the authentication server as part of the initial authorization request. This ensures that the user is redirected directly to the specified IdP.
```

> The `idp_hint` is a custom parameter supported by the application to improve the login experience in multi-Identity Provider (IdP) scenarios.

Reviewer: ```suggestion
The `idp_hint` is a custom parameter supported by the application to improve the login experience in multi-identity provider (IdP) scenarios.
```

> It allows a client application (e.g., a custom welcome page) to suggest which IdP should be preselected during the authentication flow.

Reviewer: ```suggestion
It allows a client application (for example, a custom welcome page) to suggest which IdP should be preselected during the authentication flow.
```

1. A client custom application appends the `idp_hint` parameter to the login URL:

```
https://{{< domain-c8y >}}/apps/public/login/index.html?idp_hint=google
```

> 2. The {{< product-c8y-iot >}} application extracts the value of `idp_hint` (e.g., `google`).

Reviewer: ```suggestion
2. The {{< product-c8y-iot >}} application extracts the value of `idp_hint` (for example, "google").
```

> 3. During the OAuth2/OIDC authorization request, this value is mapped and passed to the underlying Auth Server.

Reviewer: ```suggestion
3. During the OAuth2/OIDC authorization request, this value is mapped and passed to the underlying authentication server.
```

> ##### Example with Keycloak

Reviewer: ```suggestion
#### Example with Keycloak {#example-with-keycloak}
```

Keycloak supports [an identity provider hinting feature](https://www.keycloak.org/docs/latest/server_admin/index.html#_client_suggested_idp) using the `kc_idp_hint` parameter.  

> If the `idp_hint` value is `google`, and authorization request headers are configured with key: `kc_idp_hint`, value `${idp_hint}`, {{< product-c8y-iot >}} will translate it into a Keycloak compatible parameter and forward it:

Reviewer: ```suggestion
If the `idp_hint` value is "google", and the authorization request headers are configured with key `kc_idp_hint`, value `${idp_hint}`, {{< product-c8y-iot >}} will translate it into a Keycloak compatible parameter and forward it:
```
